src/default_impls.rs

```diff
@@ -99,11 +99,7 @@ mod strings {
     }
     impl DeepSizeOf for CString {
         fn deep_size_of_children(&self, _: &mut Context) -> usize {
-            // This may cause a length check at runtime, but that
-            // doesn't seem avoidable.  This assumes that the allocation
-            // is the exact length of the string and the added null
```

Comment:
in practice this assumption is true since the inner repr of `CString` is a `Box<[u8]>` which is acquired by calling `Vec::into_boxed_slice` which documents that excess capacity is dropped.